### PR TITLE
Backport ansible fixes

### PR DESCRIPTION
**Pull request by KevinFCormier**
https://github.com/open-cluster-management/backlog/issues/17720

### CODE REVIEW

File: tests/cypress/support/views.js [hunk 1]
```diff
@@ -1670,11 +1675,13 @@ const verifyHistoryPage = (mode, failuresExpected) => {
...
-    cy.get('.ansible-history-table').within(() => {
-      cy.get('svg[fill="#c9190b"]').should('have.length', failuresExpected)
-    })
   }
+  // failed jobs not available when Ansible is mocked
+  // else {
+  //   cy.get('.ansible-history-table').within(() => {
+  //     cy.get('svg[fill="#c9190b"]').should('have.length', failuresExpected)
+  //   })
+  // }
```
Comment (KevinFCormier) [inline tests/cypress/support/views.js:1684]: On release-2.4/main branches, there is a bug and failuresExpected is always 0.

File: tests/cypress/support/views.js [hunk 2]
```diff
@@ -1638,7 +1639,7 @@ export const action_verifyHistoryPageWithMock = (uName) => {
                 message: 'Awaiting next reconciliation',
                 name: 'policy-pod-1111-policy-automation-once-hd5xz',
                 started: '2021-06-03T14:45:53.237671Z',
-                status: 'successful',
+                status: successful ? 'successful' : 'error',
```
Comment (ChunxiAlexLuo) [inline tests/cypress/support/views.js:1642]: Good idead

**Review by ChunxiAlexLuo — APPROVED** [review]
/lgtm

**Pull request merged**